lib/developer_portal/app/controllers/developer_portal/login_controller.rb

```diff
@@ -31,19 +31,29 @@ def create
       self.current_user = @user
       create_user_session!
       flash[:notice] = @strategy.new_user_created? ? 'Signed up successfully' : 'Signed in successfully'
+
+      AuditLogService.call("Signed in: #{current_user.id}/#{current_user.username} #{current_user.first_name} #{current_user.last_name}")
+
       redirect_back_or_default(@strategy.redirect_to_on_successful_login)
     elsif @strategy.redirects_to_signup?
       @strategy.on_signup(session)
+
       redirect_to @strategy.signup_path(params), notice: 'Successfully authenticated, please complete the signup form'
     else
+      attempted_cred = auth_strategy_is_internal? ? params.fetch(:username, nil) : params.fetch(:token, nil)
+      AuditLogService.call("Login attempt failed: #{request.internal_host} - #{attempted_cred}")
+
       render_login_error(@strategy.error_message)
     end
   end
 
   def destroy
+    user = current_user
     logout_killing_session!
     destroy_user_session!
 
+    AuditLogService.call("Signed out: #{user.id}/#{user.username} #{user.first_name} #{user.last_name}")
```

Comment:
Similar here, we audit UserSession. Maybe no need to also log that a user signed out.